qa/python_models/instance_init_del/util.py

```diff
@@ -0,0 +1,168 @@
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+
+#
+# Helper functions for reading/writing state to disk
+#
+
+
+def read_int_from_txt_file(filename):
+    full_path = os.path.join(os.environ["MODEL_LOG_DIR"], filename)
+    try:
+        with open(full_path, mode="r", encoding="utf-8", errors="strict") as f:
+            txt = f.read()
+    except FileNotFoundError:
+        txt = "0"
+    return int(txt)
+
+
+def write_int_to_txt_file(filename, number):
+    full_path = os.path.join(os.environ["MODEL_LOG_DIR"], filename)
+    txt = str(number)
+    with open(full_path, mode="w", encoding="utf-8", errors="strict") as f:
+        f.write(txt)
+
+
+#
+# Functions for communicating initialize and finalize count between the model
+# and test
+#
+
+
+def get_count(filename):
+    count = read_int_from_txt_file(filename)
+    return count
+
+
+def inc_count(filename):
+    count = read_int_from_txt_file(filename)
+    count += 1
+    write_int_to_txt_file(filename, count)
+    return count
+
+
+def reset_count(filename):
+    count = 0
+    write_int_to_txt_file(filename, count)
+    return count
+
+
+def get_initialize_count():
+    return get_count("instance_init_del_initialize_count.txt")
+
+
+def inc_initialize_count():
+    return inc_count("instance_init_del_initialize_count.txt")
+
+
+def reset_initialize_count():
+    return reset_count("instance_init_del_initialize_count.txt")
+
+
+def get_finalize_count():
+    return get_count("instance_init_del_finalize_count.txt")
+
+
+def inc_finalize_count():
+    return inc_count("instance_init_del_finalize_count.txt")
+
+
+def reset_finalize_count():
+    return reset_count("instance_init_del_finalize_count.txt")
+
+
+#
+# Functions for communicating varies of delay (in seconds) to the model
+#
+
+
+def get_initialize_delay():
```

Comment:
How about `get_delay(kind)`?
where kind can be initialize/infer/finalize.

Reply:
Yes. Consolidated: [Util function consolidation](https://github.com/triton-inference-server/server/pull/5635/commits/cd601a939a97331ad1ad499cb2eb183e1eeea5bb)